packages/foundation/src/createComponent.tsx

```diff
@@ -47,7 +47,7 @@ export function createComponent<
   const { factoryOptions = {} } = options;
   const { defaultProp } = factoryOptions;
 
-  const result: React.FunctionComponent<TComponentProps> = (
+  const ResultComponent: React.FunctionComponent<TComponentProps> = (
```

Comment:
This fixes a false positive where the rule complains about hooks being called in a function which doesn't appear to be a component or hook (this function is actually a component, it just didn't look like a component before).